### PR TITLE
[BUZZOK-29304] DataRobot docs search and fetch url tools

### PR DESCRIPTION
**Pull request by az-dr**
# RATIONALE

Adding DataRobot docs tools of
- Search agentic docs
- Fetch DataRobot docs page (can be any DataRobot docs page, doesn't have to be agentic docs page)

We're looking to use the search agentic docs tool in an updated agent example for `af-component-agent`. On first tool invocation, the process will be:

- Fetch and parse https://docs.datarobot.com/en/docs/sitemap.xml to obtain all agentic AI docs URLs
- Fetch the contents of all agentic AI docs pages (there are 28 in total) and store the title, URL, and text content of each page
- Use TF-IDF to "search" based on a user query and return most relevant docs pages. Optional parameter `max_results` can be used to cap number of pages returned - by default is set to 3.

Docs pages are cached for 24 hour interval before needing to be fetched again - I think this is acceptable as docs pages probably don't change all that often.

The timing of the entire cycle (from fetching and parsing pages to search) tends to be 4 to 5 seconds on my home Internet and local MacBook although can be longer. An example 10 test runs:

```
dr_docs performance benchmark  (10 runs each)
============================================================

[1/2] Index build benchmark (fetches sitemap + all agentic-ai pages)
  index build run  1/10: 4.478s
  index build run  2/10: 4.600s
  index build run  3/10: 4.520s
  index build run  4/10: 4.479s
  index build run  5/10: 5.358s
  index build run  6/10: 4.523s
  index build run  7/10: 4.716s
  index build run  8/10: 4.473s
  index build run  9/10: 4.752s
  index build run 10/10: 5.246s

  Results: avg=4.715s  min=4.473s  max=5.358s
```

## CHANGES
There are two sets of tool functions that effectively provide the same functionality: 
- DataRobot MCP tool functions annotated with `@dr_mcp_tool`
- Standalone tool functions that can be wrapped as LangChain (or other framework) tool. 

The DataRobot MCP tool functions invoke the standalone tool functions. The former are intended for use in MCP server when it exists, and the latter are intended for usage outside of MCP server - LangChain tools are useable by all the frameworks we support (LangGraph, CrewAI, LlamaIndex, Nemo Agent Toolkit).

---

> [!NOTE]
> **Medium Risk**
> Introduces new network-facing behavior (bulk HTTP fetch + HTML parsing) executed inside the tool path, which may affect latency and reliability depending on external docs availability.
> 
> **Overview**
> Adds new **DataRobot documentation tools** (`dr_docs`) that (1) build an in-memory TF‑IDF index from the public `docs.datarobot.com` sitemap + Agentic AI pages (cached for 24h) to return relevant pages, and (2) fetch and extract text from any English DataRobot docs URL.
> 
> Integrates the tools into the MCP tool registry/config with a new `enable_dr_docs_tools` flag and `MCP_CLI_CONFIGS` option, adds `beautifulsoup4` to the `drmcp` dependency group, bumps version to `0.5.8`, and includes unit + E2E tests covering search, fetch, and edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 89c38f38c86ceab70ff2b2e722552dd44851cd5a. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/datarobot_genai/drmcp/test_utils/tool_base_ete.py [hunk 1]
```diff
@@ -179,12 +179,12 @@ async def _run_test_with_expectations(
...
-                ), (
-                    f"Should have called {tool_call.tool_name} tool with the correct parameters, "
-                    f"but got: {tool_call.parameters}"
-                )
+                expected_params = test_expectations.tool_calls_expected[i].parameters
+                if expected_params is not None:
+                    assert tool_call.parameters == expected_params, (
+                        f"Should have called {tool_call.tool_name} tool with the correct "
+                        f"parameters, but got: {tool_call.parameters}"
+                    )
```
Comment (rabih-datarobot) [inline src/datarobot_genai/drmcp/test_utils/tool_base_ete.py]: did you run the integration and acceptance tests locally? we don't have ci for them.
You need .env in your root directory with:
```
DATAROBOT_API_TOKEN=....
DATAROBOT_ENDPOINT=https://staging.datarobot.com/api/v2
```
run:
`task drmcp-integration` and `task drmcp-acceptance`

File: src/datarobot_genai/drmcp/tools/dr_docs/langchain_tools.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        "query": "MCP server setup",
+        "max_results": 5
+    })
+"""
+
+import logging
+from typing import Annotated
+from typing import Any
+
+from langchain_core.tools import tool
```
Comment (rabih-datarobot) [inline src/datarobot_genai/drmcp/tools/dr_docs/langchain_tools.py]: We are looking into moving drmcp out of the this library, this brings `langchain_core` dependency that we don't need, if we can remove this decorator and try to added in the client that would be great to avoid adding agent dependencies into the mcp

File: src/datarobot_genai/drmcp/tools/dr_docs/tools.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    fetch_datarobot_doc_page as _fetch_datarobot_doc_page,
+)
+from datarobot_genai.drmcp.tools.dr_docs.langchain_tools import (
+    search_datarobot_agentic_docs as _search_datarobot_agentic_docs,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dr_mcp_tool(tags={"datarobot", "docs", "documentation", "search"})
```
Comment (rabih-datarobot) [inline src/datarobot_genai/drmcp/tools/dr_docs/tools.py]: We changed this decorator just today to `dr_mcp_integration_tool` instead

File: src/datarobot_genai/drmcp/tools/dr_docs/tools.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    Note:
+        - The index covers only https://docs.datarobot.com/en/docs/agentic-ai/ (~28 pages).
+    """
+    result = await _search_datarobot_agentic_docs.ainvoke(
+        {"query": query, "max_results": max_results}
+    )
+    return ToolResult(structured_content=result)
+
+
+@dr_mcp_tool(tags={"datarobot", "docs", "documentation", "fetch", "read"})
```
Comment (rabih-datarobot) [inline src/datarobot_genai/drmcp/tools/dr_docs/tools.py]: same here, lets change the name of this decorator

**Review by rabih-datarobot — COMMENTED** [review]
Well done, looks good, left few comments, something else to note, once this merged, you need to do some updates to the af mcp component to complement these changes, lets sync once this is merged and I'll show you what is left to do

File: src/datarobot_genai/drmcp/tools/clients/dr_docs.py [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+Searches the DataRobot agentic-ai documentation section without requiring
+any API keys. Parses the sitemap.xml to discover agentic-ai documentation
+page URLs, fetches their full HTML content concurrently at index-build time,
+and builds a TF-IDF index from real page titles and body text. The index is
+cached in memory for fast repeated queries.
+
+Only pages under https://docs.datarobot.com/en/docs/agentic-ai/ are indexed,
+keeping the corpus small (~28 pages) so the full index can be built in a few
+seconds at server startup.
+"""
```
Comment (rabih-datarobot) [inline src/datarobot_genai/drmcp/tools/clients/dr_docs.py:40]: @az-dr could you also add this note here, we try to keep the tools as push down approach, but for this case we can't do it now until we move to the global-mcp.
```
TODO: Pre-built index (push index build out of MCP):
Build the TF-IDF index in a separate process (workload api), then store it (serialized file or blob). 
Sitemap fetch, 28 HTTP requests, HTML parsing, tokenization, and index construction run outside the MCP.
In the MCP:
Load pre-built index from file/URL or bundled resource.
search_docs = load index if needed + TF-IDF search.
Same for fetch_page_content if cache page contents in the artifact.
No sitemap or bulk HTML fetch, no parsing/tokenizing in the MCP. 
Only loads that artifact and runs search in memory.
```

File: src/datarobot_genai/drmcp/tools/dr_docs/langchain_tools.py (hunk 2 above)
Comment (az-dr) [reply src/datarobot_genai/drmcp/tools/dr_docs/langchain_tools.py]: roger that, I will remove the langchain decorator and leave the underlying function instead

File: src/datarobot_genai/drmcp/test_utils/tool_base_ete.py (hunk 1 above)
Comment (az-dr) [reply src/datarobot_genai/drmcp/test_utils/tool_base_ete.py]: yes I ran acceptance tests locally and saved llm responses, which look as expected

File: src/datarobot_genai/drmcp/tools/clients/dr_docs.py (hunk 5 above)
Comment (az-dr) [reply src/datarobot_genai/drmcp/tools/clients/dr_docs.py:40]: thank you, will do that

File: src/datarobot_genai/drmcp/test_utils/tool_base_ete.py (hunk 1 above)
Comment (az-dr) [reply src/datarobot_genai/drmcp/test_utils/tool_base_ete.py]: I don't recall modifying this file actually, I don't think I meant to change it - I've reverted this change

File: src/datarobot_genai/drmcp/tools/dr_docs/local_tools.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+"""Standalone (non-MCP) DataRobot Agentic AI documentation search tools.
+
+Provides standalone tool functions that can be wrapped as LangChain or other framework
+(e.g. LlamaIndex) tools. No MCP server or FastMCP dependency is required.
+
+Example use with LangGraph::
+
+    from datarobot_genai.drmcp.tools.dr_docs import search_datarobot_agentic_docs
+    from langchain_core.tools import tool
+    search_tool = tool(search_datarobot_agentic_docs)
```
Comment (rabih-datarobot) [inline src/datarobot_genai/drmcp/tools/dr_docs/local_tools.py:24]: Nice one, yeah that is better to keep dependencies clean in our side, also thanks for renaming the file

File: pyproject.toml
Comment (rabih-datarobot) [inline pyproject.toml]: @az-dr this was reverted recently, please rebase your branch before merging

**Review by rabih-datarobot — COMMENTED** [review]
@az-dr looks great thanks so much, great work, please rebase your branch and we good to merge, ping me when that is done and will approve, very impressive

**Review by rabih-datarobot — APPROVED** [review]
Great stuff

**Comment by az-dr** [comment]
Hello @smagee-robot , I would like to add you as docs reviewer on this PR

File: tests/drmcp/unit/dr_docs_tools/test_tools.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    async def test_search_returns_results(self) -> None:
+        """Test that search returns structured results on success."""
+        mock_results = [
+            {
+                "url": "https://docs.datarobot.com/en/docs/agentic-ai/index.html",
+                "title": "Agentic AI Overview",
+                "description": "Introduction to DataRobot agentic AI",
+            },
+            {
+                "url": "https://docs.datarobot.com/en/docs/agentic-ai/agentic-glossary.html",
```
[the remark this answers is not in the document]
Comment (az-dr) [reply tests/drmcp/unit/dr_docs_tools/test_tools.py:34]: yes these URLs are hard-coded - they are used as mock results so the pages are not actually being accessed in this test, the test is only to ensure that the mock results are processed properly

File: src/datarobot_genai/drmcp/tools/clients/dr_docs.py [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+                df = self._df.get(token, 0)
+                if tf > 0 and df > 0:
+                    idf = math.log(num_docs / df)
+                    score += tf * idf
+            if score > 0:
+                scored.append((score, page))
+
+        scored.sort(key=lambda x: x[0], reverse=True)
+        return [page for _, page in scored[:max_results]]
+
```
[the remark this answers is not in the document]
Comment (az-dr) [reply src/datarobot_genai/drmcp/tools/clients/dr_docs.py:198]: I'm pretty sure this is how TF-IDF is intended to work
